fortran95.html.markdown

```diff
@@ -0,0 +1,445 @@
+    real, pointer :: p    !declare a pointer.
+
+    integer, parameter :: LP = selected_real_kind(20)
+    real (kind = LP) :: d    !long precision variable.
+
+    ! WARNING: initialising variables during declaration causes problems
+    ! in functions since this automatically implies the 'save' attribute
+    ! whereby values are saved between function calls. In general, separate
+    ! declaration and initialisation code except for constants!
+    
+    
+    ! Strings
+    ! =======
+
+    character :: a_char = 'i'
+    character (len = 6) :: a_str = "qwerty"
+    character (len = 30) :: str_b
+    character (len = *), parameter :: a_long_str = "This is a long string."
+    !can have automatic counting of length using (len=*) but only for constants.
+    
+    str_b = a_str // " keyboard"    !concatenate strings using // operator.
+
+
+    ! Assignment & Arithmetic
+    ! =======================
+
+    Z = 1    !assign to variable z declared above (case insensitive).
+    j = 10 + 2 - 3
+    a = 11.54  /  (2.3 * 3.1)
+    b = 2**3    !exponentiation
+    
+    !an aside:
+    print *, b    !print the variable 'b' to the command line (see I/O section).
+    
+
+    ! Control Flow Statements & Operators
+    ! ===================================
+
+    ! Single-line if statement
+    if (z == a) b = 4  !condition always need surrounding parentheses.
+
+    if (z /= a) then !z not equal to a
+    ! Other symbolic comparisons are < > <= >= == /=
+      b = 4
+    else if (z .GT. a) then !z greater than a
+    ! Text equivalents to symbol operators are .LT. .GT. .LE. .GE. .EQ. .NE.  
+      b = 6
+    else if (z <  a) then !'then' must be on this line.
+      b = 5 !execution block must be on a new line.
+    else
+      b = 10
+    end if !end statement needs the 'if' (or can use 'endif').
+
+
+    if (.NOT. (x < c .AND. v >= a .OR. z == z)) then   !boolean operators.
+      inner: if (.TRUE.) then    !can name if-construct.
+        b = 1
+      endif inner    !then must name endif statement.
+    endif
+
+
+    i = 20
+    select case (i)
+      case (0)    !case i == 0
+        j=0
+      case (1:10)    !cases i is 1 to 10 inclusive.
+        j=1
+      case (11:)    !all cases where i>=11
+        j=2
+      case default
+        j=3
+    end select
+
+
+    month = 'jan'
+    ! Condition can be integer,logical or character.
+    ! Select constructions can also be named.
+    monthly: select case (month)
+      case ("jan")
+         j = 0
+      case default
+         j = -1
+    end select monthly
+
+
+    do i=2,10,2    !loops from 2 to 10 (inclusive) in increments of 2.
+      innerloop: do j=1,3    !loops can be named too.
+        exit    !quits the loop.
+      end do innerloop
+    cycle    !jump to next loop iteration.
+    enddo
+
+    
+    ! Goto statement exists but it is heavily discouraged though.
+    goto 10    
+    stop 1000   !stops code immediately (returning specified condition code).
```

Comment:
The condition code is an integer so I just chose 1000 at random. Maybe there's more to it than that, I'll check.